### PR TITLE
Typescript error squashing: AmazonPayWallet.tsx

### PR DESCRIPTION
**Pull request by GHaberis**
## What are you doing in this PR?

This PR contains Typescript error fixes in `AmazonPayWallet.tsx`. As part of this I've introduced types for the `AmazonPaymentsObject` in the `contributionsLandingReducer.ts`, which includes types for the `Wallet` and `Consent` widgets.

### CODE REVIEW

File: support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts [hunk 1]
```diff
@@ -65,9 +65,59 @@ export type StripeCardFormData = {
 	createPaymentMethod: ((clientSecret: string | null) => void) | null;
 	handle3DS: ((clientSecret: string) => Promise<Stripe3DSResult>) | null; // For single only
 };
+type Wallet = {
+	bind: (ID: string) => void;
+};
+type Consent = {
+	bind: (ID: string) => void;
+};
+export type BaseWalletConfig = {
```
Comment (tomrf1) [inline support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts:74]: should we prefer `interface` over `type` in TS?

Comment (GHaberis) [reply support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts:74]: I went with `type` as the the other types in `contributionsLandingReducer.ts` used `type`, but if the convention is to prefer interface I can switch these (and update the rest of the definitions in this module).

**Comment by GHaberis** [comment]
[Typescript error squashing: AmazonPayWallet.tsx](https://trello.com/c/DLa1xX3V/182-typescript-error-squashing-amazonpaywallettsx)

File: support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts [hunk 2]
```diff
@@ -16,60 +15,112 @@ import type {
...
+	}) => void;
+}
+interface WalletConstructor {
+	new (baseWalletConfig: WalletConfig): Wallet;
+}
+export interface ConsentConfig {
+	sellerId: string;
+	amazonBillingAgreementId: string;
+	onReady: (billingAgreementConsentStatus: {
+		getConsentStatus: () => 'true' | 'false';
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts:100]: Ahhh I remember this one - who needs booleans anyway! Cheers Amazon!

**Review by TomPretty — APPROVED** [review]
Looks good to me 👍

**Pull request merged**